docs/en/operations/settings/settings.md

```diff
@@ -1777,6 +1777,19 @@ Possible values:
 
 Default value: 0.
 
+## distributed_directory_monitor_split_batch_on_failure {#distributed_directory_monitor_split_batch_on_failure}
+
+Enables/disables splitting batches on failures.
```

Comment:
But from what i understand it don't split anything, it just disable batching when batched insert fails. So it will not split the data into the pieces smaller than was inserted by user.

Reply:
>But from what i understand it don't split anything, it just disable batching when batched insert fails.

I thought about splitting at first, but there is no good choice on how to do this, so simply disabling should be good enough.

> So it will not split the data into the pieces smaller than was inserted by user.

The idea was to split batches in terms of files, i.e. make the batch from smaller number of files, not to split the INSERT block in one file.

Reply:
Worth mentioning that in docs + highlighting that it adjusts `distributed_directory_monitor_batch_inserts` behavior. 

But doing that during the retry sounds a bit harmful - we may go in a bad pattern when we always first try to insert the data as a bigger batch, and later as a smaller batch. In effect always sending the same data twice (even if the second try is always successful). Maybe we can try to introduce something like a dynamically adjusted max_batch_size - i.e. when we see errors we decrease the max batch size, when we see no errors for a longer time - try to increase it / return to normal value?

BTW: I've seen the problem 'in the wild' too.

Reply:
> dynamically adjusted max_batch_size 

Dynamic batch size is a completely separate thing.

Right now it just "disable batching" for the batches that cannot be handled, disabling batching globally looks odd (even based on some heuristics)

And detect proper batch size is also not that simple.

To me just disabling the batching looks enough, since it is enough for server to make a progress and in case of every batch is too huge to send at once then setup should be fixed instead.

Reply:
>Worth mentioning that in docs + highlighting that it adjusts distributed_directory_monitor_batch_inserts behavior.

Actually there was already a sentence about this:

>but sending files from that batch one by one may succeed INSERT.

But anyway I've extended the documentation paragraph, hope that this will clarify something for users.